_docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Plugin/AssemblyBuild/OnPageNavigationBuild.php

```diff
@@ -63,12 +74,19 @@ public function build(array &$build, EntityInterface $entity, EntityViewDisplayI
           '#theme' => 'item_list',
           '#list_type' => 'ul',
           '#items' => $nav_items,
-          '#attributes' => ['class' => 'on-page-nav-list'],
+          '#attributes' => [
+            'class' => [
+              'on-page-nav-list',
+              'pf-c-list', 'pf-m-inline ',
+              'rhd-u-no-bullet',
+            ],
+          ],
         ];
       }
     }
 
-    $build['#cache']['tags'][] = $build['#parent']['entity']->getEntityTypeId() . ":" . $build['#parent']['entity']->id();
+    $cacheableMetadata->applyTo($build);
```

Comment:
Just for reference / my own sanity in the review, this is what we're adding to the `#cache`. This is what we would expect. I am not 100% certain that we _need_ to add assemblies that are placed above the OnPageNavigation assembly, in this case the https://localhost/assembly/32715/view, but I don't think that it really hurts necessarily. Do you have thoughts @staceymosier about whether or not it is worth looking into excluding assemblies that are not rendered in the OnPageNavigation in this PR?

```
array(5) {
  ["tags"]=>
  array(7) {
    [0]=>
    string(14) "assembly:32715"
    [1]=>
    string(14) "assembly:33055"
    [2]=>
    string(14) "assembly:33075"
    [3]=>
    string(14) "assembly:33095"
    [4]=>
    string(14) "assembly:33155"
    [5]=>
    string(14) "assembly:33175"
    [6]=>
    string(14) "assembly:33555"
  }
  ["contexts"]=>
  array(0) {
  }
  ["max-age"]=>
  int(-1)
  ["keys"]=>
  array(4) {
    [0]=>
    string(11) "entity_view"
    [1]=>
    string(8) "assembly"
    [2]=>
    string(5) "33035"
    [3]=>
    string(7) "default"
  }
  ["bin"]=>
  string(6) "render"
}
```

Reply:
I pushed a change to include the parent entity (node) as a cache tag. 
I think keeping all the available assemblies as cache tags is necessary, in the event that they are rearranged on the node.  Although right now the node itself can suffice as the only cache tag,  the rest of the assemblies here are necessary in the case that the 'assembly update triggers node update' design changes in the future.